### ISSUE TITLE
Support for CR3 not working

### ISSUE THREAD

**Issue by han-k59** [ncruces/dcraw#1]
I tried using DCRAW 0.9.28 for the CR3 format. No error messages but the output file contains black only. I used the RAW CR3 samples files from:

https://www.photographyblog.com/reviews/canon_eos_m50_review/sample_images
(scroll further fown for CR3 files)

**Comment by ncruces** [comment]
Main branch of this repo tracks Dave Coffin's `dcraw`.
If it's not supported [there](https://dechifro.org/dcraw/), it's not supported here.

**If this is something you can fix yourself**, the process should be: fork this, make the change, and upstream the change, by sending an email to Dave.

**If this is something you can't fix yourself**: please send Dave an email with bug report, samples.

**I'll happily merge back any changes Dave makes.**

---
Now, I'll be honest, I haven't been very successful at up streaming my fixes.

If you fail to upstream your changes (as I have), do a PR, and I'll consider merging them into a devel branch with my own changes.

**Comment by diegocr** [comment]
any update on this by any chance?

Also, it's just me or looks like there is some code missing:

https://github.com/ncruces/dcraw/blob/eafa02db3fd8aab533144b24baf4ef89a45b9ede/dcraw.c#L6886-L6888

...which may or may not be related to the broken cr3 decoding.

**Comment by ncruces** [comment]
No. I also do not have the time to pursue this.

I'll keep the repo synched with upstream, and maintain my own changes (one of which is an ugly hack, but neither of which are on master).

If you find this useful as a base, feel welcome to fork my repo.

As I said, I'll also consider merging small, easy to maintain changes in a development branch, and keep nagging Dave to get them upstream, but my success so far has been zero.

**Comment by diegocr** [comment]
How about filing a https://www.bountysource.com/ for this to be done? and ideally, something Dave himself would do so that we do support its work (?) but anybody could take it.

**Comment by ncruces** [comment]
I'm not opposed to that. To be very honest, I just don't want to be project manager for this. This is still very much Dave's effort, and I've had zero contact with him so far.

I just found it useful to have a repo with full history on GitHub, so ported it here.

**Comment by diegocr** [comment]
Ah, got it, i thought you may was a friend of him, given for example i have no idea where his git/svn repository is/was :)

**Comment by han-k59** [comment]
I have modified my program to use LibRaw. They support CR3. That's a better solution for the future since it's an active project.

**Comment by ncruces** [comment]
@diegocr There's no git repo. He has some RCS files that include history (not even an SVN repo). I used a tool to convert those into this.

**Comment by ncruces** [comment]
@han-k59 Right, that's a good point.

I'm only using `dcraw` because I wanted something out-of-process. I could build on LibRaw but that's more effort.

**Comment by han-k59** [comment]
I have returned to DCRAW. I'm not happy  about Libraw showing unused sensor areas. For my purpose "stacking of astronomical images" these unused areas are  causes problems.

I have forked the code and added a modification to add shutter/exposure time, isospeed, camera model and some other info to the comment section of the .pgm file header. Retrieving the exposure time is highly important for my program.  I hope @ncruces is willing to merge this change.

CR3 is not working. I hope somebody can find a fix. I'm very  unfamiliar with C and can't help here. Prefer object Pascal.
